js/bitfinex2.js

```diff
@@ -1440,8 +1440,15 @@ module.exports = class bitfinex2 extends bitfinex {
             }
             tag = this.safeString (data, 3);
             type = 'withdrawal';
-        } else {
+        } else if (transactionLength === 22) {
```

Comment:
Comparing the size of the array to 22 doesn't look totally right to me. Is this intentional?

Reply:
It was intentional in my opinion (even though not directly related to this issue, because in my mind, each object passed to the 'parseXYZ' functions should have its specific deterministic 'if' block, otherwise, it may not be clear what object (from same class or even its derived class) are passed to 'parseXYZ'.
in this specific case, as bitfinex returns array-responses (without keys) the only reliable way to target the 'fetchTransaction' record was to compare to array-size.

But if you think it's not ok, then sure, lets remove and be there only 'else'. (I'll leave this part to you during merge, so I won't re-commit).